modules/docs/mdx/8.0-UPGRADE-GUIDE.mdx

```diff
@@ -41,6 +42,173 @@ encounter any issues or use cases that we've missed.
 
 ## Components
 
+### Component Deprecations
+
+#### Deprecation Types
+
+There are two types of deprecations: soft and hard.
+
+##### Soft Deprecation
+
+A soft-deprecated component is still available with its full functionality, but it will have been
+renamed with a prefix to indicate its soft-deprecated status. It will also include a console warning
+announcing its deprecation. This warning will only be triggered on the component's initial render.
+
+Soft-deprecated types and utilities will also be renamed but generally will not trigger a console
+warning.
+
+##### Hard Deprecation
+
+A hard-deprecated component or package is no longer available. You will need to follow the method
+prescribed in our upgrade guide to update your application. Please
+[reach out](https://github.com/Workday/canvas-kit/issues/new?labels=bug&template=bug.md) to our team
+directly if you need additional help.
+
+#### Drawer
+
+We are [soft deprecating](#soft-deprecation) `Drawer`. It has been renamed to `DeprecatedDrawer`.
+Similarly, `DrawerProps` has been renamed to `DeprecatedDrawerProps`, `DrawerHeader` to
+`DeprecatedDrawerHeader` and `DrawerHeaderProps` to `DeprecatedDrawerHeaderProps`. You may continue
+to use this component, but note that we plan to [hard-deprecate](#hard-deprecation) it in Canvas Kit
+v9.
+
+🤖 The codemod will rename `Drawer`, `DrawerProps`, `DrawerHeader` and `DrawerHeaderProps` to their
+deprecated names:
+
+```tsx
+// v7
+import {Drawer, DrawerHeader} from '@workday/canvas-kit-labs-react/drawer';
+
+<Drawer
+  header={
+    <DrawerHeader
+      closeIconAriaLabel="Close"
+      onClose={() => console.log('onClose callback')}
+      title="Deprecated Drawer Header"
+    />
+  }
+/>;
+
+// v8
+import {DeprecatedDrawer, DeprecatedDrawerHeader} from '@workday/canvas-kit-labs-react/drawer';
+
+<DeprecatedDrawer
+  header={
+    <DeprecatedDrawerHeader
+      closeIconAriaLabel="Close"
+      onClose={() => console.log('onClose callback')}
+      title="Deprecated Drawer Header"
+    />
+  }
+/>;
+```
+
+#### Menu (preview)
+
+We are [soft deprecating](#soft-deprecation) `Menu`. It has been renamed to `DeprecatedMenu`.
+Similarly, `MenuProps` has been renamed to `DeprecatedMenuProps`, `MenuState` to
+`DeprecatedMenuState`, `MenuItem` to `DeprecatedMenuItem` and `MenuItemProps` to
+`DeprecatedMenuItemProps`. You may continue to use this component, but note that we plan to
+[hard-deprecate](#hard-deprecation) it in Canvas Kit v9.
```

Comment:
I think we should mention something in here that people can use the menu in the main package

Reply:
I just added reminder that deprecated component can be replaced. This message can be changed during doc update work.